### PR TITLE
fix: #1517 #1495 Should be displayed warning message when enters the space in all search fields

### PR DESCRIPTION
**Pull request by ant066**
# Pull request checklist

## Does this close any currently open issues?

fixes: #1517 #1495

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #1517 #1495

## What is the new behavior?

- Will displayed warning message when no field be filled
Screen Shot 2020-06-05 at 3 41 03 PM
Screen Shot 2020-06-05 at 3 41 11 PM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/marketplace/src/utils/__tests__/object.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { cleanObject } from '../object'
```
Comment (phmngocnghia) [inline packages/marketplace/src/utils/__tests__/object.ts:1]: Could we rename this func to something more meaningful. E.g.
`removeKeyWhereValueIsFalsy`

Comment (ant066) [reply packages/marketplace/src/utils/__tests__/object.ts:1]: I think we should not use a long name for the function. Let me add a function description.

File: packages/marketplace/src/components/pages/admin-apps/admin-apps.tsx [hunk 2]
```diff
@@ -125,15 +127,11 @@ export const generateColumns = ({ dispatch, setDataDeleteModal, deleteModalData
   ]
 }
 
-export const refreshForm = (onSubmit: Function, resetForm: Function) => () => {
-  resetForm()
-  onSubmit({ appName: '', companyName: '', developerName: '' })
+export const refreshForm = history => () => {
+  history.push('apps')
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-apps/admin-apps.tsx:131]: Please avoid hardcoded strings, `apps` should be moved to `routes.ts`

**Comment by phmngocnghia** [comment]
@ant066  conflict

File: packages/marketplace/src/components/pages/admin-apps/admin-apps.tsx (hunk 2 above)
Comment (ant066) [reply packages/marketplace/src/components/pages/admin-apps/admin-apps.tsx]: I changed it. Use Routes config now

File: packages/marketplace/src/components/pages/admin-apps/admin-apps.tsx [hunk 3]
```diff
@@ -207,15 +207,15 @@ export type FormValues = {
...
+export const handleOnSubmit = history => (formValues: FormValues, { setStatus }) => {
+  const cleanedValues = cleanObject(formValues)
+
+  if (isEmptyObject(cleanedValues)) {
+    setStatus('Please enter at least one search criterion')
+    return
+  }
+
+  const queryString = stringifyObjectIntoQueryString({ ...cleanedValues, page: 1 })
   history.push(`apps?${queryString}`)
```
Comment (nphivu414) [inline packages/marketplace/src/components/pages/admin-apps/admin-apps.tsx:219]: Please use `Routes.ADMIN_APPS` here too

**Pull request merged**